local_planner/src/local_planner.cpp

```diff
@@ -396,32 +408,38 @@ void LocalPlanner::getStateAndTwistInput() {
     stand_pose_ << current_state_[0], current_state_[1], current_state_[5];
   }
 
-  // If it's going to walk, use latest states
-  if (cmd_vel_[0] != 0 || cmd_vel_[1] != 0 || cmd_vel_[5] != 0) {
+  // Set initial condition for forward integration
+  Eigen::Vector2d support_center;
+  support_center.setZero();
+  for (int i = 0; i < num_feet_; i++) {
+    support_center.x() +=
+        robot_state_msg_->feet.feet[i].position.x / ((double)num_feet_);
+    support_center.y() +=
+        robot_state_msg_->feet.feet[i].position.y / ((double)num_feet_);
+  }
+
+  // Walk if velocity commanded, current velocity exceeds threshold, or too far
+  // from center of support
+  if (cmd_vel_.norm() > 1e-1 || current_state_.segment(6, 2).norm() > 1e-1 ||
```

Comment:
Could make each of these conditions a lambda function for better readability

Reply:
Used a bool for this.